### PR TITLE
[TACHYON-1346] optionally wait for process to finish/die

### PR DESCRIPTION
**Pull request by abridgett**
this is useful when running under cluster managers etc.

This PR superceeds #2159

### CODE REVIEW

File: integration/bin/tachyon-mesos.sh [hunk 1]
```diff
@@ -7,7 +7,37 @@ SCRIPT_DIR="$(cd "$(dirname "$0")"; pwd)"
...
+      echo -e "$Usage"
+      exit 1
+      ;;
+  esac
+done
+
+shift $((OPTIND-1))
+
 MESOS_MASTER_ADDRESS="$1"
+if [ ! "$MESOS_MASTER_ADDRESS" ]; then
```
Comment (jsimsa) [inline integration/bin/tachyon-mesos.sh]: [[ ]]

Comment (jsimsa) [inline integration/bin/tachyon-mesos.sh]: @{MESOS_MASTER_ADDRESS}

File: integration/bin/tachyon-mesos.sh [hunk 2]
```diff
@@ -7,7 +7,37 @@ SCRIPT_DIR="$(cd "$(dirname "$0")"; pwd)"
...
+  echo -e "$Usage"
```
Comment (jsimsa) [inline integration/bin/tachyon-mesos.sh]: ${Usage}

File: integration/bin/tachyon-mesos.sh [hunk 3]
```diff
@@ -7,7 +7,37 @@ SCRIPT_DIR="$(cd "$(dirname "$0")"; pwd)"
 source "${SCRIPT_DIR}/common.sh"
 TACHYON_FRAMEWORK_JAVA_OPTS="${TACHYON_FRAMEWORK_JAVA_OPTS:-${TACHYON_JAVA_OPTS}}"
 MESOS_LIBRARY_PATH="${MESOS_LIBRARY_PATH:-/usr/local/lib}"
+
+Usage="Usage: tachyon-mesos.sh [-hw] MESOS_MASTER_ADDRESS
++MESOS_MASTER_ADDRESS is of the form 'mesos.example.com:5050'
++
++-w  Wait for process to finish before returning
```
Comment (jsimsa) [inline integration/bin/tachyon-mesos.sh]: s/Wait/wait

File: bin/tachyon-start.sh [hunk 4]
```diff
@@ -28,6 +28,8 @@ MOPT is one of:
 
 -N  Do not try to kill prior running masters and/or workers in all or local
 
+-w  Wait for processes to end before returning
```
Comment (jsimsa) [inline bin/tachyon-start.sh]: s/Wait/wait

**Comment by jsimsa** [comment]
@abridgett thanks for the PR ... I left some minor comments ... overall this LGTM

@apc999 please take a quick look

File: bin/tachyon-start.sh (hunk 4 above)
Comment (apc999) [inline bin/tachyon-start.sh]: I think -N has comment "Do not ...", starting with capitalized "D", so may be fine

**Comment by apc999** [comment]
+1 LGTM

**Comment by abridgett** [comment]
I've fixed the syntax (sorry!)
Corrected the case (also the line above which I'd based it from).    I think it's a little more readable capitalised so if you like I'll change them all.

**Comment by apc999** [comment]
@abridgett Thanks. The PR looks good now.

I will merge it

**Comment by jsimsa** [comment]
LGTM

**Pull request merged**